lib/compat/wordpress-6.9/site-editor-redirect.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/**
+ * Maps old site editor urls to the new updated ones.
+ *
+ * @since 6.9.0
+ * @access private
+ *
+ * @global string $pagenow The filename of the current screen.
+ *
+ * @return string|false The new URL to redirect to, or false if no redirection is needed.
+ */
+function gutenberg_get_site_editor_redirection_6_9() {
+	global $pagenow;
+
+	if ( 'site-editor.php' !== $pagenow ) {
+		return false;
+	}
+
+	// The following redirects are for the new permalinks in the site editor.
+	if ( isset( $_GET['p'] ) && preg_match( '/^\/wp_template\/([a-zA-Z0-9-]+\/\/[a-zA-Z0-9-]+)$/', $_GET['p'], $matches ) ) {
```

Comment:
Any point in using `$_REQUEST` instead? Also, probably a good place to use a different delimiter for the PCRE pattern:

```suggestion
	if ( isset( $_GET['p'] ) && preg_match( '_^/wp_template/([a-zA-Z0-9-]+//[a-zA-Z0-9-]+)$_', $_GET['p'], $matches ) ) {
}
```

Reply:
As a friendly bonus, I also suggest adding a comment above with the overall shape of the redirection, to make scanning the code easier. Something like:

```js
// /wp_template/tt5//home -> /wp_registered_template/tt5//home
```

Reply:
Why should we use `$_REQUEST`? Sure, we can use a different pattern and add a comment

Reply:
Whoops, the suggestion introduced an extra }

Reply:
Oops, sorry. :)

> Why should we use `$_REQUEST`?

No good reason comes to mind, but the fact that `gutenberg_get_site_editor_redirection` did made me wonder. Is there any scenario where we could expect a POST request to need redirection? Put differently: could it hurt to use `$_REQUEST`?

Reply:
Ok, the suggested pattern doesn't work 🤔

Reply:
I switched it to use #

Reply:
Ooops, I chose poorly: there are obviously underscores in the pattern. But my point still stands:

```php
if ( preg_match( '#^/wp_template/([a-zA-Z0-9-]+//[a-zA-Z0-9-]+)$#', $input, $matches ) ) {
```

Reply:
> I switched it to use #

Jinx :) 
(Also, when will GitHub finally properly update comment threads in real time...?)

Reply:
Done